### PR TITLE
Use PIN_I2S_ defines if they exist in board pin definition.

### PR DESCRIPTION
**Pull request by DatanoiseTV**
Make the usage more user friendly and use PIN_I2S_ pin definitions if they are defined in the board pins_arduino.h.

For example: https://github.com/DatanoiseTV/arduino-pico/blob/master/variants/datanoisetv_picoadk/pins_arduino.h#L43

### CODE REVIEW

**Review by earlephilhower — CHANGES_REQUESTED** [review]
I like your idea, but this really isn't the way to do it with the way the class is architected now.

These defaults should be se set in the `I2S::I2S` constructor and not the `::begin`.  See https://github.com/earlephilhower/arduino-pico/blob/master/libraries/I2S/src/I2S.cpp#L26-L39

**Comment by DatanoiseTV** [comment]
@earlephilhower Would this work for you?
```cpp
I2S::I2S(PinMode direction) {
    _running = false;
    _bps = 16;
    _writtenHalf = false;
    _isOutput = direction == OUTPUT;
#ifdef PIN_I2S_BCLK
    _pinBCLK = PIN_I2S_BCLK;
#else
    _pinBCLK = 26;
#endif

#ifdef PIN_I2S_DOUT
    if (_isOutput == OUTPUT) {
        _pinDOUT = PIN_I2S_DOUT;
    }
#endif

#ifdef PIN_I2S_DIN
    if (_isOutput == INPUT) {
        _pinDOUT = PIN_I2S_DIN;
    }
#endif
    _freq = 48000;
    _arb = nullptr;
    _cb = nullptr;
    _buffers = 8;
    _bufferWords = 16;
    _silenceSample = 0;
}
```

**Comment by earlephilhower** [comment]
That looks good, yes!

**Comment by DatanoiseTV** [comment]
@earlephilhower The weird thing is just the _pinDOUT confusion. It would be good to support both INPUT and OUTPUT and combined mode for I2S audio codec. Do you have an idea how to solve it?

**Comment by earlephilhower** [comment]
The way the state machine works it's not possible to do both shift directions in the same PIO SM.  So you need 2 separate state machines if you want bidirectional.  (They can, of course, be running completely different bit widths, frequencies, etc.)

**Comment by earlephilhower** [comment]
@DatanoiseTV can you update the PR with what we discussed above?  That way it can be included in the next release this Friday or Saturday...

**Comment by DatanoiseTV** [comment]
> @DatanoiseTV can you update the PR with what we discussed above? That way it can be included in the next release this Friday or Saturday...

Sorry, done just now.

File: libraries/I2S/src/I2S.cpp [hunk 1]
```diff
@@ -27,11 +27,26 @@ I2S::I2S(PinMode direction) {
...
     _writtenHalf = false;
+    _isOutput = direction == OUTPUT;
     _pinBCLK = 26;
     _pinDOUT = 28;
+#ifdef PIN_I2S_BCLK
+    _pinBCLK = PIN_I2S_BCLK;
+#endif
+
+#ifdef PIN_I2S_DOUT
+    if (_isOutput == OUTPUT) {
```
Comment (earlephilhower) [inline libraries/I2S/src/I2S.cpp:38]: I think you mean "`if (_isOutput)` since this is a boolean...

File: libraries/I2S/src/I2S.cpp [hunk 2]
```diff
@@ -27,11 +27,26 @@ I2S::I2S(PinMode direction) {
...
+        _pinDOUT = PIN_I2S_DOUT;
+    }
+#endif
+
+#ifdef PIN_I2S_DIN
+    if (_isOutput == INPUT) {
```
Comment (earlephilhower) [inline libraries/I2S/src/I2S.cpp:44]: Same as above...`if (!_isOutput)`...

**Review by earlephilhower — CHANGES_REQUESTED** [review]

**Comment by earlephilhower** [comment]
Thx, but I think there's a couple minor things as noted.  OTW looks good!

**Review by earlephilhower — APPROVED** [review]
Thanks!

**Pull request merged**